### PR TITLE
Added Repeat Weekly Feature

### LINKED ISSUE #20: Repeat Weekly feature
**Describe the feature you'd like**
Currently, when a user creates a profile it is active for only one week and after that, it does not work.
A Repeat weekly toggle should be present in NewProfileFragment which should allow the user to repeat the profile for consecutive weeks till they decide to stop it by deleting it.

**Screenshots**
scrnli_05_10_2020_01-46-05

**Would you like to work on the issue?**

- [ ] Yes
- [ ] No
- Other: Open for all.

### PR DESCRIPTION
**Pull request by coder2699**
Fixes #20

Changes:  Added repeat weekly feature

### CODE REVIEW

**Comment by coder2699** [comment]
@rob729 Please have a look at this PR.  One more thing, since PeriodicTimeWorkRequest is calculating 7*24hours, so it was not practically possible to check it.

So what I did was, in order to check the working, I changed the PeriodicTimeWorkRequest to 16 minutes (as it should be atleast 15mins), and it was working fine. So if you want to check the app functioning it can be done this way.

**Review by rob729 — CHANGES_REQUESTED** [review]
Switches in the middle of the screen are not looking good.
Try by making text and the switch shift to the start and end of the screen.
Share the screenshot after making the changes.

**Comment by coder2699** [comment]
@rob729 Please have a look

**Comment by rob729** [comment]
@coder2699 Looks better. Go ahead with this change.

**Comment by coder2699** [comment]
> @coder2699 Looks better. Go ahead with this change.

@rob729 I have pushed the code with the asked changes.

File: app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt [hunk 1]
```diff
@@ -243,12 +245,21 @@ class NewProfileFragment : Fragment() {
...
         val profileData = workDataOf(Pair("ActiveProfileId", profile.profileId), Pair("Profile_Name", profile.name), (Pair("VibrateKey", profile.vibSwitch)), (Pair("EndTimeKey", etime)))
 
-        val startAlarmRequest = OneTimeWorkRequest.Builder(StartAlarm::class.java)
-            .addTag(profile.profileId.toString())
-            .setInputData(profileData)
-            .setInitialDelay(c.timeInMillis - System.currentTimeMillis(), TimeUnit.MILLISECONDS)
-            .build()
-        context?.let { WorkManager.getInstance(it).enqueue(startAlarmRequest) }
+        if (repeatWeeklySwitch.isChecked) {
+            val startAlarmRequest = PeriodicWorkRequest.Builder(StartAlarm::class.java, 7, TimeUnit.DAYS)
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt]: A little code improvement change.
val startAlarmRequest =   if (repeatWeeklySwitch.isChecked) {.......
and take context?.let { WorkManager.getInstance(it).enqueue(startAlarmRequest) } out of if block.

File: app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt [hunk 2]
```diff
@@ -262,14 +273,22 @@ class NewProfileFragment : Fragment() {
 
         timeCheck(c)
         val profileData = workDataOf(Pair("Profile_Name", profile.name))
+        if (repeatWeeklySwitch.isChecked) {
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt]: same comment as above.

**Comment by rob729** [comment]
@coder2699 Can you share an APK in which the PeriodicTimeWorkRequest interval is 16 minutes for testing purpose?

**Comment by coder2699** [comment]
> @coder2699 Can you share an APK in which the PeriodicTimeWorkRequest interval is 16 minutes for testing purpose?

@rob729 Yes, you can use the link below to get app apk for testing purpose.
https://drive.google.com/file/d/13axyQkGjpR16MmtusjvEef67LNCWH2Gz/view?usp=sharing

**Review by rob729 — CHANGES_REQUESTED** [review]
Show a repeat icon in the details fragment so that the user knows it is a recurring profile.

The above icon has to be used with accent color instead of black color. 
Place this icon on the right of the vibrate/silent mode icon.
The profile title, vibrate icon, and repeat icon should be in a linear layout and that linear layout should be center aligned.
Also, reduce the gap between profile title and start time text by 25%.
Also, while editing the profile the value of repeat weekly is not taken into consideration. Fix it.

**Comment by coder2699** [comment]
@rob729 Please check

**Comment by rob729** [comment]
@coder2699 Share a screenshot in case of a non-repeating profile also.

**Comment by coder2699** [comment]
> @coder2699 Share a screenshot in case of a non-repeating profile also.


@rob729 This is SS for non repeating.

File: app/src/main/java/com/rob729/quiethours/Fragments/DetailsFragment.kt [hunk 3]
```diff
@@ -59,6 +61,8 @@ class DetailsFragment : BottomSheetDialogFragment() {
             binding.end.text = "${setTimeString(args.ehr)}:${setTimeString(args.emin)}"
             if (args.vibSwitch) binding.audioMode.setImageResource(R.drawable.vibration)
             else binding.audioMode.setImageResource(R.drawable.mute)
+            if (args.repeatWeekly)binding.repeatWeeklyIcon.visibility = VISIBLE
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Fragments/DetailsFragment.kt]: change to binding.repeatWeeklyIcon.visibility = if (args.repeatWeekly) ....

Comment (coder2699) [reply app/src/main/java/com/rob729/quiethours/Fragments/DetailsFragment.kt]: @rob729 Done. Please have a look.

**Review by rob729 — APPROVED** [review]

**Comment by rob729** [comment]
give @coder2699 150 points

**Pull request merged**